### ISSUE TITLE
gitignore, config.js, package.json, jspm install, jspm update

### ISSUE THREAD

**Issue by jpray** [jspm/jspm-cli#970]
After using jspm (and loving it!) with a group of developers, it seems that config.js should possibly be split between two files.  One that is committed and would likely contain baseUrl, transpiler, babelOptions, paths, meta.  And one that is gitignored and would likely contain paths and maybe bundles.  Paths is the big one that seems like it should be ignored.  Basically if running jspm install or update is modifying config.js, I would like those changes to be ignored to avoid merge conflicts and also because if it is committed and pulled by another developer, it isn't accurate compared to the state of their specific jspm_packages folder (which is gitignored of course).  Alternatively, might the recommended approach be to use git filters or something fancy like that to ignore part of config.js?

I also wish that updating the version of a package in package.json and running jspm install would update the package.  I've learned that one needs to run jspm update to do this, but it seems that if the minimum version declared in package.json is not met by the local version of the package, then jspm install should update the package.  Thanks in advance for considering these items...

**Comment by guybedford** [comment]
@jpray glad you're enjoying the project. `jspm install` is a version lock - it uses the `map` config in the `config.js` file to replicate the exact jspm_packages structure. This is why having the `map` config checked in is important for reproducibility of installs. To update a dependency after editing the package.json use `jspm update` rather.

`paths` configuration is only a couple of lines, so I'm not sure what benefit is gained by ignoring it. `bundles` configuration can certainly be included in a separate config and this is expected really for custom build setups to output the bundle configuration in its own bootstrap layer. It is certainly encouraged to create a manual setup for this sort of thing and not use the `jspm bundle -i` feature in jspm.

**Comment by jpray** [comment]
Oops, just realized I said `paths` when I actually meant `map` is the big one that I thought should be ignored.  But thinking of config.js as a version lock, I see why it should be committed. And what you said about keeping the bundles config in separate file instead of injecting it into config.js makes sense too.  

I think our workflow problems can be fixed by having the same developer that added the new dependency or (updated the version of a dependency) also run jspm install or jspm update and check in the updated config.js.  We were having everyone run jspm update separately, but this way they can just run jspm install and that shouldn't alter config.js. Thanks!

**Comment by peteruithoven** [comment]
I'm still struggling with this, on the project I'm working on we usually have a couple of feature branches that people work on individually for multiple days. This means that the packages and package versions we use start to diverge pretty quickly. 
When we try to merge these we get multiple conflicts in the jspm.config.js file. I'd need to manually check that I select the map / package configurations that match the precise versions I have installed on my computer? And then still I'm not sure what to do with the packages I don't have installed yet. Our `jspm.config.js` is currently `1937` lines, without bundles.
Is there any way to regenerate `jspm.config.js`, based on the package.json?

**Comment by peteruithoven** [comment]
What further complicates fixing conflicts is that the order seems random. A `react` entry in `map` can be 200 lines lower from branch to the other.

**Comment by jpray** [comment]
I haven't switched to jspm 0.17 yet so not sure if this has been addressed there, but the ordering not being consistent has been a pain point for me too, mainly the order for the bundles config.  If it hasn't already been addressed, hopefully it can be made so that running a bundle command will produce the same output if the underlying config/files are the same.

Just a couple thoughts regarding your other points.  If I get conflicts on the `map` or `bundles` part of the config, I don't hesitate to blow those two sections away and rerun jspm install or jspm update or the appropriate bundling commands to regenerate them.  The two exceptions for this are if I have manually modified the `map` part or if I am close to a release date and I want to be conservative about updating the versions of any dependencies.  But generally the most churn on dependencies is at the beginning of an development cycle so blowing away and regenerating `map` has worked pretty well for me.
